GH-9694: Remove duplicate SocketChannel initialization

anthologia opened the pull request:
Fixes: https://github.com/spring-projects/spring-integration/issues/9694
Issue link: https://github.com/spring-projects/spring-integration/issues/9694

- Remove duplicate socketChannel initialization
- Properly close the SocketChannel when connection fails
- Prevents socket resource leakage during connection retries

Commit:
GH-9694: Remove duplicate SocketChannel initialization

pivotal-cla commented:
@anthologia Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=9696)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=9696) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@anthologia Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=9696)!

artembilan commented on spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioClientConnectionFactory.java:
```diff
@@ -113,6 +115,11 @@ protected TcpConnectionSupport buildNewConnection() {
 		}
 		catch (IOException e) {
+			try {
+				socketChannel.close();
```
Don't we need to check if `socketChannel != null` first?

anthologia replied:
Ah right, I missed the null check. Thanks for catching that!

artembilan commented on spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioClientConnectionFactory.java:
```diff
@@ -113,6 +115,11 @@ protected TcpConnectionSupport buildNewConnection() {
 		catch (IOException e) {
+			try {
+				socketChannel.close();
+			} catch (IOException e2) {
```
Our code style is to have `catch` on a new line.
You can run locally `./gradlew :spring-integration-ip:check`.

anthologia replied:
Checked the style guide and fixed it accordingly. Thanks for the feedback :)

anthologia committed:
GH-9694: Add null check when closing SocketChannel

Fixes: #9694
Issue link: https://github.com/spring-projects/spring-integration/issues/9694

- Add null check before closing socketChannel in catch block
- Prevents potential NullPointerException during error handling

artembilan commented:
@anthologia ,
thank you for contribution; looking forward for more!
